src/CalcManager/CEngine/scicomm.cpp

```diff
@@ -16,6 +16,8 @@
 #include "Header Files/CalcEngine.h"
 #include "Header Files/CalcUtils.h"
 
+#include <iterator>
```

Comment:
Move standard headers to the pch so they can be compiled once and used throughout the project.

Reply:
I'm not in favour of that. PCH are commonly hard to work with and include unnecessary items that you wouldn't otherwise have.

Reply:
Why is it hard to work with?  The STL headers aren't changing between compilation runs.  These projects already use PCH and so this change is abnormal.  I'm open to a community discussion for whether we should use PCH or not, but it's working fine right now and I don't see an immediate reason to move away.

Reply:
^ Sorry, meant standard library headers in general

Reply:
The problematic part is many (most?) build systems don't support them adequately and forcing users to compile everything C++ and the project has to offer for each translation unit is a huge overkill.

Knowing how PCHs work, I would rather imagine the project to be set up so that only the minimal required set of headers is included, then the build system supporting PCH injects it into translation unit.

Reply:
I'm happy to have a community discussion on how we can properly support pch or move to no pch.  That conversation should not be decided by this PR.  Until we have that conversation, we should follow the standards of the codebase, and the convention here is to keep standard headers in the pch.  In the new discussion, we can compare compilation times, how they are improved by pch, and what the impact would be for us if we move off of pch and update source files to only include what is needed (as you are suggesting here).

Reply:
I have updated it as requested, will file a ticket for discussing PCH later.

Reply:
I filed a ticket for reworking PCHs: https://github.com/Microsoft/calculator/issues/324